frontend/packages/dev-console/src/components/import/image-search/ImageStream.tsx

```diff
@@ -0,0 +1,192 @@
+    const imageStreamsList = {};
+    const isBuilderImageNamespace = ns === builderImagesNamespace.Openshift;
+    const imgStreams = isBuilderImageNamespace
+      ? (normalizeBuilderImages(imageStreams) as NormalizedBuilderImages)
+      : (imageStreams as K8sResourceKind[]);
+    _.each(imgStreams, (img: BuilderImage | K8sResourceKind) => {
+      const { name, namespace } = isBuilderImageNamespace
+        ? (img as BuilderImage).obj.metadata
+        : (img as K8sResourceKind).metadata;
+      if (namespace === ns) {
+        imageStreamsList[name] = name;
+      }
+    });
+    return imageStreamsList;
+  };
+  const getImageStreamTags = (name: string) => {
+    const tags = {};
+    const imageStream = _.find(imageStreams, ['metadata.name', name]);
+    name &&
+      !_.isEmpty(imageStream) &&
+      _.each(imageStream.status.tags, ({ tag }) => {
+        tags[tag] = tag;
+      });
+    return tags;
+  };
+  React.useEffect(() => {
+    if (imageNamespace.value !== selectedProject) {
+      setselectedProject(imageNamespace.value);
+      setFieldValue('imageStream.image', '');
+      setFieldValue('imageStream.tag', '');
+      setFieldValue('isi', initialValues.isi);
+    }
+    if (imageName.value !== selectedName) {
+      setselectedName(imageName.value);
+      setFieldValue('imageStream.tag', '');
+    }
+    if (imageTag.value !== selectedTag) {
+      setselectedTag(imageTag.value);
+      imageTag.value !== '' && searchImageTag();
+    }
+  }, [
+    imageNamespace,
+    imageName,
+    imageTag,
+    selectedProject,
+    selectedName,
+    selectedTag,
+    setFieldValue,
+    initialValues.isi,
+    searchImageTag,
+  ]);
+
+  return (
+    <>
+      <div className="row">
+        <div className="col-lg-4 col-md-4 col-sm-4 col-xs-12">
+          <DropdownField
+            name="imageStream.namespace"
+            label="Projects"
+            items={projectList}
+            title="Project"
+            fullWidth
+            required
+          />
+        </div>
+        <div className="col-lg-4 col-md-4 col-sm-4 col-xs-12">
+          <DropdownField
+            name="imageStream.image"
+            label="ImageStreams"
+            items={getImageStreamByNamespace(values.imageStream.namespace)}
+            title="ImageStreams"
+            fullWidth
+            disabled={values.imageStream.namespace === ''}
+            required
+          />
+          <div className="odc-imagestream-separator">/</div>
+        </div>
+        <div className="col-lg-4 col-md-4 col-sm-4 col-xs-12">
+          <DropdownField
+            name="imageStream.tag"
+            label="Tag"
+            items={getImageStreamTags(values.imageStream.image)}
+            title="Tag"
+            disabled={values.imageStream.image === ''}
+            fullWidth
+            required
+          />
+          <div className="odc-imagestream-separator">:</div>
+        </div>
+      </div>
+      {values.registry === registryType.Internal &&
```

Comment:
This condition will never be tru since value of registry in case of internal is being set as `"Image name from internal registry"`.